lib/src/native/realm_core.dart

```diff
@@ -226,13 +226,15 @@ class _RealmCore {
           _realmLib.realm_sync_config_set_error_handler(syncConfigPtr, _realmLib.addresses.realm_dart_sync_error_handler_callback, errorHandlerUserdata.cast(),
               _realmLib.addresses.realm_dart_userdata_async_free);
 
-          if (config.clientResetHandler is! ManualRecoveryHandler) {
+          if (config.clientResetHandler.onBeforeReset != null) {
```

Comment:
OK! You have changed this, so I don't need to make additional PR, as you suggested in Client Reset PR.

Reply:
yes no need

Reply:
OK! You have changed this, so I don't need to make additional PR, as you suggested in Client Reset PR.